shell/platform/windows/flutter_windows_view_unittests.cc

```diff
@@ -0,0 +1,163 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <iostream>
+#include <vector>
+
+#include "flutter/shell/platform/common/json_message_codec.h"
+#include "flutter/shell/platform/embedder/test_utils/proc_table_replacement.h"
+#include "flutter/shell/platform/windows/flutter_windows_engine.h"
+#include "flutter/shell/platform/windows/flutter_windows_texture_registrar.h"
+#include "flutter/shell/platform/windows/flutter_windows_view.h"
+#include "flutter/shell/platform/windows/testing/engine_modifier.h"
+#include "flutter/shell/platform/windows/testing/mock_window_binding_handler.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace flutter {
+namespace testing {
+
+constexpr uint64_t kScanCodeKeyA = 0x1e;
+constexpr uint64_t kVirtualKeyA = 0x41;
+
+namespace {
+
+// A struct to use as a FlutterPlatformMessageResponseHandle so it can keep the
+// callbacks and user data passed to the engine's
+// PlatformMessageCreateResponseHandle for use in the SendPlatformMessage
+// overridden function.
+struct TestResponseHandle {
+  FlutterDesktopBinaryReply callback;
+  void* user_data;
+};
+
+static const bool test_response = false;
+
+constexpr uint64_t kKeyEventFromChannel = 0x11;
+constexpr uint64_t kKeyEventFromEmbedder = 0x22;
+static std::vector<int> key_event_logs;
+
+std::unique_ptr<std::vector<uint8_t>> keyHandlingResponse(bool handled) {
+  rapidjson::Document document;
+  auto& allocator = document.GetAllocator();
+  document.SetObject();
+  document.AddMember("handled", test_response, allocator);
+  return flutter::JsonMessageCodec::GetInstance().EncodeMessage(document);
+}
+
+// Returns an engine instance configured with dummy project path values, and
+// overridden methods for sending platform messages, so that the engine can
+// respond as if the framework were connected.
+std::unique_ptr<FlutterWindowsEngine> GetTestEngine() {
```

Comment:
Would it make sense to have these in a separate file so that they can be used in other tests?

Reply:
I would but for now I don't think they share enough logic. Many test files mock engine but in different ways respectively. It can be a different task in the future.